packages/in_app_purchase/in_app_purchase_android/android/src/main/java/io/flutter/plugins/inapppurchase/MethodCallHandlerImpl.java

```diff
@@ -374,6 +378,41 @@ private void updateCachedSkus(@Nullable List<SkuDetails> skuDetailsList) {
     }
   }
 
+  private void launchPriceChangeConfirmationFlow(String sku, MethodChannel.Result result) {
+    if (activity == null) {
+      result.error(
+          "ACTIVITY_UNAVAILABLE",
+          "launchPriceChangeConfirmationFlow is not available. "
+              + "This method must be run with the app in foreground.",
+          null);
+      return;
+    }
+    if (billingClientError(result)) {
+      return;
+    }
+    assert billingClient != null;
```

Comment:
I know it `assert` is used on more places in the code base (and in other plugins as well) but I recently learned that in most situations `assert` doesn't work on Android development (see also https://stackoverflow.com/a/6176529/5167831 and https://stackoverflow.com/a/8164195/5167831). With this in mind the recommended solution would be to throw an `AssertionError`:

```suggestion
    if (billingClient == null) {
      throw new AssertionError("The `billingClient` should not be null.");
    }
```

Reply:
The `if (billingClientError(result))` is already doing `if (billingClient == null)`, so this is only so the analyzer doesn't warn it can be null, because it can't be. Changing this doesn't really add anything, but maybe we could write a test for `billingClientError` that it returns true if billingClient == null.

This is the billingClientError method:
```
  private boolean billingClientError(MethodChannel.Result result) {
    if (billingClient != null) {
      return false;
    }

    result.error("UNAVAILABLE", "BillingClient is unset. Try reconnecting.", null);
    return true;
  }
```

Reply:
Then let's add a comment to the `assert` line that it is only there to silence the analyser. Maybe something like:
```suggestion
    // Note that assert doesn't work on Android (see https://stackoverflow.com/a/6176529/5167831 and https://stackoverflow.com/a/8164195/5167831)
    // and that this assert is only added to silence the analyser. The actual null check 
    // is handled by the `billingClientError()` call. 
    assert billingClient != null;
```